src/app/app.routes.ts

```diff
@@ -1,3 +1,8 @@
 import { Routes } from '@angular/router';
+import { LoginComponent } from './components/login/login.component';
+import { SignupComponent } from './components/signup/signup.component';
 
-export const routes: Routes = [];
+export const routes: Routes = [
+    { path: 'login', component: LoginComponent },
+    { path: 'signup', component: SignupComponent },
```

Comment:
Just check that you have routes for: if there is an error (sane default no match), and if the url string is empty that it redirects somewhere. this might be in a different branch but I thought I'd just mention it.

Reply:
I added them in my UI branch. A page not found, and an app page for '' that can route you to either login or signup